### PR TITLE
(MODULES-11392) Add Puppet 7 to 8 upgrade test

### PR DESCRIPTION
**Pull request by mhashizume**
This commit adds a Beaker test for upgrading from Puppet 7 to Puppet 8.

### CODE REVIEW

**Review by joshcooper — APPROVED** [review]

**Comment by mhashizume** [comment]
Passed an experimental pipeline here: https://jenkins-platform.delivery.puppetlabs.net/view/__experimental%20automatic/job/experimental_auto_puppetlabs-puppet-agent-module_intn-sys_pa-acceptance_7-nightly_to_8-nightly-adhoc/6/

File: acceptance/helpers.rb [hunk 1]
```diff
@@ -87,8 +87,9 @@ def require_master_collection(args)
       if args.is_a?(Symbol)
         collection = args.to_s
 
-        # puppet_collection_for doesn't know about nightly collections
-        unless collection.include?(server_collection)
+        # Dirty, temporary workaround for when we have puppet8 agent nightlies but not server nightlies
+        # Once we have puppet8 server releases, remove the second conditional
+        unless collection.include?(server_collection) || collection.include?('nightly')
```
Comment (joshcooper) [inline acceptance/helpers.rb:92]: Should this check for `puppet8-nightly` specifically or is it a general issue for puppet7-nightly too?

**Comment by joshcooper** [comment]
LGTM

File: acceptance/helpers.rb (hunk 1 above)
Comment (mhashizume) [reply acceptance/helpers.rb]: Good catch, I've updated the commit and limited it to `puppet8-nightly`

**Pull request merged**